protocol/scripts/deploys/helpers.ts

```diff
@@ -269,6 +270,32 @@ export async function deployAndMine<
   );
   console.log('********************\n');
 
+  // If we are on hardhat chain
+  if (network.config.chainId === 4 && process.env.ETHERSCAN_API_KEY) {
+  // Wait for a few blocks to be mined before verifying
+  await contract.deployTransaction.wait(6)
+  // Verify the contract on Etherscan
+  const response = await axios.post(
+    `https://api.etherscan.io/api?module=contract&action=verifysourcecode&apikey=${process.env.ETHERSCAN_API_KEY}`,
+    {
+      contractaddress: contract.address,
+      sourceCode: (contract as any).deployedBytecode,
+      contractname: name,
+      compilerversion: 'v0.7.0+commit.9e61f92b',
```

Comment:
most of our stuff is v0.8.0 but this can vary. You should be able to pull it from the hardhat config I believe?

Reply:
gonna get crackin on this td